server/src/main/java/org/apache/druid/client/indexing/ClientCompactionRunnerInfo.java

```diff
@@ -160,6 +163,10 @@ public static CompactionConfigValidationResult validateRollupForMSQ(
       return CompactionConfigValidationResult.failure(
           "MSQ: 'granularitySpec.rollup' must be true if 'metricsSpec' is specified"
       );
+    } else if (metricsSpec == null && isRollup != null && isRollup) {
```

Comment:
What about empty `metricsSpec`? I guess `rollup` must be false in that case too.

Reply:
Thanks. Added to the check.